### ISSUE TITLE
Time Input Fails Accessibility at Desktop Level

### ISSUE THREAD

**Issue by JakeLaCombe** [cerner/terra-core#1670]
# Bug Report

## Description
Time Input needs to be accessible.

## Steps to Reproduce
1. Go to https://engineering.cerner.com/terra-core/#/raw/tests/terra-time-input/time-input/twelve-hour/default
2. Do a lighthouse audit in desktop mode

## Additional Context / Screenshots
* Form Elements do not have associated labels. Minutes and Hours should have their own labels poentially.
* Color contrast is invalid on the mobile inputs. That needs designer feedback to correct.

## Expected Behavior
Accessible Time Picker

**Comment by bjankord** [comment]
@neilpfeiffer Can we get some color updates on this for the default theme to make it match contrast ratios.

**Comment by neilpfeiffer** [comment]
@JakeLaCombe What are the items that are currently failing with contrast ratio checks? Is it specifically the help-text for mobile inputs or is it the placeholder color for both desktop+mobile?

**Comment by JakeLaCombe** [comment]
I'm not getting any accessibility failures on the placeholder elements. I think since it's the placeholder value that no audits are being fully performed on that element. The only failures flagging are color contrast on the help texts.

As for other failures, for the 12 hour clocks, an aria-label attribute needs to be added to both the meridiem display, and the select box that contains the values of the meridiems. We'd have to work through how we want those attributes to be applied to those elements.

**Comment by bjankord** [comment]
Related: https://github.com/cerner/terra-core/issues/1840
